server/rotate.go

```diff
@@ -25,20 +25,58 @@ func (a *Agent) RotateLog() {
 		return
 	}
 
+	if _, err := a.db.ExecContext(ctx, "FLUSH LOCAL ERROR LOGS"); err != nil {
+		a.logger.Error(err, "failed to exec FLUSH LOCAL ERROR LOGS")
+		metrics.LogRotationFailureCount.Inc()
+		return
+	}
+
+	durationSeconds := time.Since(startTime).Seconds()
+	metrics.LogRotationDurationSeconds.Observe(durationSeconds)
+}
+
+// RotateLog rotates slow log files
+func (a *Agent) RotateSlowLog() {
+	ctx := context.Background()
+
+	metrics.LogRotationCount.Inc()
+	startTime := time.Now()
+
 	slowFile := filepath.Join(a.logDir, mocoagent.MySQLSlowLogName)
-	err = os.Rename(slowFile, slowFile+".0")
+	err := os.Rename(slowFile, slowFile+".0")
 	if err != nil && !os.IsNotExist(err) {
 		a.logger.Error(err, "failed to rotate slow query log file")
 		metrics.LogRotationFailureCount.Inc()
 		return
 	}
 
-	if _, err := a.db.ExecContext(ctx, "FLUSH LOCAL ERROR LOGS, SLOW LOGS"); err != nil {
-		a.logger.Error(err, "failed to exec FLUSH LOCAL LOGS")
+	if _, err := a.db.ExecContext(ctx, "FLUSH LOCAL SLOW LOGS"); err != nil {
+		a.logger.Error(err, "failed to exec FLUSH LOCAL SLOW LOGS")
 		metrics.LogRotationFailureCount.Inc()
 		return
 	}
 
 	durationSeconds := time.Since(startTime).Seconds()
 	metrics.LogRotationDurationSeconds.Observe(durationSeconds)
 }
+
+// RotateLogIfSizeExceeded rotates log files if it exceeds rotationSize
+func (a *Agent) RotateLogIfSizeExceeded(rotationSize int64) {
+	errFile := filepath.Join(a.logDir, mocoagent.MySQLErrorLogName)
+	errFileStat, err := os.Stat(errFile)
```

Comment:
Also check for the existence of files.

Reply:
When the file doesn't exists, os.Stat() returns error.
So I think checking existence of the file is not necessary.